### PR TITLE
[XGboost] Update Documentation

### PR DESCRIPTION
**Pull request by amogkam**
Updates Documentation to match xgboost_ray README

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/xgboost-ray.rst [hunk 1]
```diff
@@ -131,6 +134,80 @@ setting this explicitly.
...
+- Store data as `float32` or less. More precision is often
+  not needed, and keeping data in a smaller format will
+  help reduce peak memory usage for initial data loading.
+- Pass the `dtype` when loading data from CSV. Otherwise,
+  floating point values will be loaded as `np.float64`
+  per default, increasing peak memory usage by 33%.
+
+Placement Strategies
+--------------------
+`xgboost_ray` leverages Ray's Placement Group API (https://docs.ray.io/en/master/placement-group.html)
```
Comment (richardliaw) [inline doc/source/xgboost-ray.rst:190]:
```suggestion
`xgboost_ray` leverages :ref:`Ray's Placement Group API <ray-placement-group-examples-ref>`)
```

File: doc/source/xgboost-ray.rst [hunk 2]
```diff
@@ -131,6 +134,80 @@ setting this explicitly.
...
+to implement placement strategies for better fault tolerance.
+
+By default, a SPREAD strategy is used for training, which attempts to spread all of the training workers
+across the nodes in a cluster on a best-effort basis. This improves fault tolerance since it minimizes the
+number of worker failures when a node goes down, but comes at a cost of increased inter-node communication
+To disable this strategy, set the `USE_SPREAD_STRATEGY` environment variable to 0. If disabled, no
```
Comment (richardliaw) [inline doc/source/xgboost-ray.rst:196]: For each of these keywords, can you use two ticks instead of one?

Comment (amogkam) [reply doc/source/xgboost-ray.rst]: Ah right

**Comment by amogkam** [comment]
Piggybacking a small mlflow typo fix in this PR as well.

**Comment by richardliaw** [comment]
Test failing? `/home/travis/build/ray-project/ray/doc/source/xgboost-ray.rst:157:Unexpected indentation.`

**Comment by amogkam** [comment]
@richardliaw can you merge this please. Doc test is passing.

**Review by richardliaw — APPROVED** [review]
Thanks for the ping!

**Pull request merged**